### PR TITLE
Add OpenTofu module examples for blog posts

### PR DESCRIPTION
**Pull request by brikis98**
## Description

These are examples that are used in a blog post to announce Boilerplate. I figured it made sense to add the official versions of these examples to this repo, where we could also have automated tests for them.

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [ ] Update the docs.
- [ ] Run the relevant tests successfully, including pre-commit checks.
- [ ] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [ ] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes (draft)

Added / Removed / Updated [X].

### CODE REVIEW

**Comment by brikis98** [comment]
OK, tests are passing, so this is ready for review.

**Comment by brikis98** [comment]
@denis256 Could I get a review on this PR? It also includes a file with `{` in the name, so it may need to be merged into your https://github.com/gruntwork-io/boilerplate/pull/164 PR.

**Comment by denis256** [comment]
Taking a look

File: examples/for-learning-and-testing/terraform-test/{{ .ModuleName | snakecase }}_test.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package test
```
Comment (denis256) [inline examples/for-learning-and-testing/terraform-test/{{ .ModuleName | snakecase }}_test.go]: Having a file with `{{` in git will break the git clone in Windows... 

So far I thought about generation of files at runtime in "unix" tests:
* have generic file name like "module_test.go.template"
* in "unix" test copy the test directory to temp dir
* rename file to "{{..." 
* run tests

I can handle this in my windows CICD branch

**Comment by brikis98** [comment]
Actually, hold up on reviewing for one minute! I need to rename a few things.

**Review by denis256 — APPROVED** [review]

File: examples/for-learning-and-testing/terraform-test/{{ .ModuleName | snakecase }}_test.go (hunk 1 above)
Comment (brikis98) [reply examples/for-learning-and-testing/terraform-test/{{ .ModuleName | snakecase }}_test.go]: Yes, I saw that. Windows is always such a PITA :-D 

The one catch with code-generating this example is that I'm linking to the example from the new blog post... So if it's code-generated, that doesn't quite work... Is there any other alternative? Does it work with Cygwin or something?

**Comment by brikis98** [comment]
Thanks for the review. Merging now so I can get the blog post published.

**Pull request merged**